vcd/datasource_vcd_catalog_media_test.go

```diff
@@ -0,0 +1,157 @@
+// Using a catalog media data source we create another catalog media
+// where the description is the first data source ID
+func TestAccVcdCatalogAndMediaDatasource(t *testing.T) {
+	var TestCatalogMediaDS = "TestCatalogMediaDS"
+	var TestAccVcdDataSourceMedia = "TestAccVcdCatalogMediaBasic"
+	var TestAccVcdDataSourceMediaDescription = "TestAccVcdCatalogMediaBasicDescription"
+
+	var media govcd.Media
+
+	var params = StringMap{
+		"Org":              testConfig.VCD.Org,
+		"Catalog":          testSuiteCatalogName,
+		"NewCatalogMedia":  TestCatalogMediaDS,
+		"OvaPath":          testConfig.Ova.OvaPath,
+		"UploadPieceSize":  testConfig.Ova.UploadPieceSize,
+		"UploadProgress":   testConfig.Ova.UploadProgress,
+		"Tags":             "catalog",
+		"CatalogMediaName": TestAccVcdDataSourceMedia,
+		"Description":      TestAccVcdDataSourceMediaDescription,
+		"MediaPath":        testConfig.Media.MediaPath,
+	}
+
+	configText := templateFill(testAccCheckVcdCatalogMediaDS, params)
+	if vcdShortTest {
+		t.Skip(acceptanceTestsSkipped)
+		return
+	}
+
+	debugPrintf("#[DEBUG] CONFIGURATION: %s", configText)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { preRunChecks(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: catalogMediaDestroyed(testSuiteCatalogName, TestCatalogMediaDS),
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config:             configText,
+				ExpectNonEmptyPlan: true,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckVcdCatalogMediaExists("vcd_catalog_media."+TestAccVcdDataSourceMedia, &media),
+					resource.TestMatchOutput("owner_name", regexp.MustCompile(`^\w*$`)),
+					resource.TestMatchOutput("creation_date", regexp.MustCompile(`^(2019|2020)-`)),
+					resource.TestCheckOutput("status", "RESOLVED"),
+					resource.TestMatchOutput("storage_profile_name", regexp.MustCompile(`(.|\s)*\S(.|\s)*`)),
+					testCheckMediaNonStringOutputs(),
+				),
+			},
+			resource.TestStep{
+				ResourceName:      "vcd_catalog_media." + TestAccVcdDataSourceMedia + "-import",
+				ImportState:       true,
+				ImportStateVerify: true,
+				ImportStateIdFunc: importStateIdByCatalogMedia(TestAccVcdDataSourceMedia),
+				// These fields can't be retrieved from catalog media data
+				ImportStateVerifyIgnore: []string{"media_path", "upload_piece_size", "show_upload_progress"},
+			},
+		},
+	})
+}
+
+func catalogMediaDestroyed(catalog, mediaName string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		conn := testAccProvider.Meta().(*VCDClient)
+		org, err := conn.GetOrgByName(testConfig.VCD.Org)
+		if err != nil {
+			return err
+		}
+		cat, err := org.GetCatalogByName(catalog, false)
+		if err != nil {
+			return err
+		}
+		_, err = cat.GetMediaByName(mediaName, false)
+		if err == nil {
+			return fmt.Errorf("catalog media %s not deleted", mediaName)
+		}
+		return nil
+	}
+}
+
+func importStateIdByCatalogMedia(objectName string) resource.ImportStateIdFunc {
```

Comment:
The function `importStateIdByCatalogItem` will return the same result. We can reuse it

Reply:
I liked to be separated-explicit, but updated